### PR TITLE
Upgrade duckdb to 0.9.2 and update dependencies

### PR DESCRIPTION
**Pull request by kenanwarren**
Running the upgrade script; also updated bindgen, arrow, and polars. Let me know if I should roll any dependency updates back.

### CODE REVIEW

File: src/raw_statement.rs [hunk 1]
```diff
@@ -143,13 +143,7 @@ impl RawStatement {
...
 
             if let Err(err) = import_arrow2_array {
-                // When array is empty, import_array_from_c returns error with message
-                // "OutOfSpec("An ArrowArray of type X must have non-null children")
-                // Therefore, we return None when encountering this error.
-                match err {
-                    arrow2::error::Error::OutOfSpec(_) => return None,
-                    _ => panic!("Failed to import arrow2 Array from C: {}", err),
-                }
+                panic!("Failed to import arrow2 Array from C: {}", err)
```
Comment (kenanwarren) [inline src/raw_statement.rs:146]: Not positive, but I don't believe this is valid comment anymore. It looks like `polars` did some housekeeping in the last couple of months with their `polars-arrow` package. It looks like the relevant bits [are here](https://github.com/pola-rs/polars/blob/main/crates/polars-arrow/src/ffi/array.rs#L20).

An aside I stumbled on this issue with some additional information on the changes:
https://github.com/jorgecarleitao/arrow2/pull/1476

Comment (kenanwarren) [reply src/raw_statement.rs]: Sorted it out, [they're using](https://github.com/pola-rs/polars/blob/d3ecfe162e8e54fc9bb2f7062629e3d38f8a6059/crates/polars-arrow/src/ffi/array.rs#L366) `PolarsError::ComputeError` now

**Comment by mlafeldt** [comment]
Thanks for the PR. Would be great to get this in (plus #238 maybe 😎) and cut a new release. 🙏

**Comment by kenanwarren** [comment]
@mlafeldt Since it moved under duckdb I'm not sure who approval and release access.

@wangfenjin Sorry to ping you, but are you still the best person to reach out to for PRs or is there someone under the DuckDB org that's taken over? Thank you again for starting this project 🙏

**Comment by wangfenjin** [comment]
Thanks for the contribution. I no longer have the write permission for this repo, I’m sure someone will handle PR

**Comment by mlafeldt** [comment]
Asked about code ownership [in Discord](https://discord.com/channels/909674491309850675/922563050173792297/1180070592158896158).

**Comment by Mytherin** [comment]
Thanks! LGTM

**Comment by szarnyasg** [comment]
> Thanks for the contribution. I no longer have the write permission for this repo, I’m sure someone will handle the PR

Hi @wangfenjin, sorry about that, that was a mistake on my end. You should have write permissions and I now restored them.

**Pull request merged**